filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -160,21 +164,24 @@ func (cim *InputManager) Create(config *common.Config) (input.Input, error) {
 	}
 
 	settings := struct {
-		ID             string        `config:"id"`
+		ID             string        `config:"id" validate:"required"`
```

Comment:
Will this prevent filebeat from starting if it was configured to use filestream inputs without an ID? This would be a breaking change wouldn't it? I'm not sure we want to do this.

Consider that there are customers with 1000+ agents that use filebeat, and they'll all just fail to start with this correct? If you are running with fleet or a large k8s deployment seeing this error is not as simple as reading one log file or some output in the terminal.

Reply:
Yes, Filebeat will fail to start and this is a breaking change, but not requiring an ID leads to data duplication with multiple inputs. I understood we were fine with this breaking change. Maybe we should discuss it a bit more.

I see 3 scenarios:
* Users run Filebeat with multiple filestream inputs without ID and have not noticed the data duplication on restart, it's a silent bug. This change will make it explicit. We can improve the error message.
* Users run standalone Filebeat and have set IDs to avoid the data duplication.
* Filebeat runs under Elastic-Agent, we're going to release together Fleet UI and Elastic-Agent to ensure all filestream inputs have an ID.

Reply:
I remember discussing that we should warn before completing deprecating filestream inputs without IDs but not much more than that. Hopefully I did not have a different opinion previously :)

The current situation results in duplication, which is an inconvenience. Preventing filebeat from starting could result in data loss (because of log rotation, for example). This is a worse outcome to me. We have escalated a bug into an outage.

I don't think we should make a breaking change here. Especially we should not make a breaking change in a minor release after a major full of breaking changes. 

I think the changes in this PR have value without breaking existing configurations. We should strongly recommend and obviously communicate the need to add IDs without breaking anything. 

I would suggest we consider automatically migrating configurations to have IDs if we can, but I don't want to derail the conversation too much with alternative solutions. We can ship this change as is without the breaking part of it.

Reply:
> I remember discussing that we should warn before completing deprecating filestream inputs without IDs but not much more than that. Hopefully I did not have a different opinion previously :)

That's what I remember as well, there is a [PR merged](https://github.com/elastic/beats/pull/30401) with the warnings (I'm not completely sure it has been released already).

> The current situation results in duplication, which is an inconvenience. Preventing filebeat from starting could result in data loss (because of log rotation, for example). This is a worse outcome to me. We have escalated a bug into an outage.

That's a very good point.

> I don't think we should make a breaking change here. Especially we should not make a breaking change in a minor release after a major full of breaking changes.

I see, it makes sense. It's tricky because that's the only way to prevent the bug :/

> I think the changes in this PR have value without breaking existing configurations. We should strongly recommend and obviously communicate the need to add IDs without breaking anything. We would need to do that anyway.

So that would apply to the IDs duplication as well, right? I'll change everything to log error then.

> I would suggest we consider automatically migrating configurations to have IDs if we can as a follow up. I don't want to derail the conversation too much with alternative solutions. We can ship this change as is without the breaking part of it.

I believe that means re-writing the configuration file, otherwise the IDs will not persist neither be stable. Anyway I agree we can have this discussion later.

Reply:
> So that would apply to the IDs duplication as well, right? I'll change everything to log error then.

Ugh, yes I suppose if there are duplicated IDs out there now then preventing startup for them is a breaking change too.

Let's upgrade the warnings to errors as you suggested for these cases, and start a follow up conversation about ways to automatically fix this problem.

At a minimum, when we release this we should be able to have support resolve filestream duplication related cases without our input after this change. We will have to make sure we put this fix in their knowledge-base, and make it really obvious in the release notes.